python/tests/test_utils.py

```diff
@@ -0,0 +1,397 @@
+import pytest
+from unittest.mock import Mock, patch
+from kubeflow.trainer.utils import utils
+from kubeflow.trainer.types import types
+from kubeflow.trainer.constants import constants
+
+
+class TestTrainerDetection:
```

Comment:
Thank you for adding unit tests!
Please can you move them under `kubeflow/trainer/types/types_test.py`, similar to our client tests: https://github.com/kubeflow/sdk/pull/17

Previously, we agreed to follow that folder structure.